src/python/gem5/resources/resource.py

```diff
@@ -624,6 +625,15 @@ def obtain_resource(
             # `exit_ok=True` ensures no exception is thrown.
             os.makedirs(resource_directory, exist_ok=True)
 
+        # if "url" starts with "file:", then we assume the resource is
+        # already on the host system and we do not need to download it.
+        # Instead, we copy the file to the resource directory.
+        if resource_json["url"].startswith("file:"):
```

Comment:
I really don't like how non-standard this is. Can we try to support the standard here https://en.wikipedia.org/wiki/File_URI_scheme ?

E.g., the following should all be valid and parsed correctly:

```
file:///home/bbruce/binary
file:/home/bbruce/binary
file://localhost/home/bbruce/binary
file:////localhost/home/bbruce/binary
```